src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2022,11 +2041,34 @@ PlanRouterQuery(Query *originalQuery,
 	 * not been called. Thus, restriction information is not avaliable and we do the
 	 * shard pruning based on the distribution column in the quals of the query.
 	 */
-	if (FastPathRouterQuery(originalQuery))
+	if (fastPathRouterQuery)
```

Comment:
I think there is some room for improvement here. The function `TargetShardIntervalForFastPathQuery ()` could cover the case when `distributionKeyValue ` exists as well.

In the end, both cases are `FastPath`

Reply:
Still valid note, in case you missed